### DISCUSSION TITLE
[PROPOSAL] Require Fortran 95

### DISCUSSION THREAD

**Discussion by derobins** [HDFGroup/hdf4#408] [Proposed Change]
### **PROBLEM**

The HDF4 Fortran wrappers are fundamentally based on Fortran 77. This has at least two problems:

* The integer type is limited to 32-bits, which is a problem since some API calls try to jam pointers into Fortran integers
* The Autotools build system contains a lot of Fortran 77 stuff and I'm afraid to touch it

### **PROPOSED FIX**

* Bring over the Fortran build system improvements from HDF5
* Require Fortran 95+
* See if we can update the HDF4 Fortran integer types to make the unsafe Fortran calls safe again (even if jamming pointers into integers for posterity is fundamentally unwise)

### **HOW WOULD USERS BE AFFECTED?**

**PRO**
➕ HDF4 would be more usable in modern Fortran environments
➕ Building HDF4 + Fortran would probably be easier
➕ It would be easier to bring Fortran improvements over from HDF5
➕ Autotools build hacks and warnings would go away
➕ Maybe we can make the Fortran wrappers safer

**CON**
➖ Anyone using Fortran 77 will no longer be able to use the Fortran wrappers
➖Some aspects of building the Fortran wrappers may change

**Comment by derobins** [comment]
This will be addressed in 4.4.0 (#424).

**Reply by mathomp4** [reply]
@derobins Sorry to reply to a zombie issue, but does this mean HDF 4.4.0 will restore Fortran support? Or just that 4.4.0 will drop FORTRAN 77?
